src/libclipper/src/datatypes.cpp

```diff
@@ -121,11 +123,7 @@ size_t FloatVector::serialize(uint8_t *buf) const {
 InputType FloatVector::type() const { return InputType::Floats; }
 
 size_t FloatVector::hash() const {
-  size_t cur_hash = 0;
-  for (const auto d : data_) {
-    cur_hash ^= std::hash<float>()(d);
-  }
-  return cur_hash;
+  return primitive_input_hash(data_);
```

Comment:
This is fine for this PR, but hashing floats is generally a "Bad Idea" (for the same reasons that floating point comparison can be tricky).

Can you add a TODO to the `FloatVector` and `DoubleVector` hash function declarations in `datatypes.hpp` noting that we should change this, and create a JIRA issue to track this as well (include the JIRA issue number in the TODO)?